goosebit/updater/manager.py

```diff
@@ -267,3 +267,8 @@ async def delete_device(dev_id: str) -> None:
         del device_managers[dev_id]
     except KeyError:
         pass
+
+
+def reset_update_manager():
```

Comment:
Yes - I am also not too happy with the global state but did not yet look deeper into it.

`force_update` likely could be replaced by setting the `update_state` of a device to `registered' or 'finished' and resetting device.fw_version. Maybe it then would be called "reset state".

Reply:
Note: there is currently also a bug when manually assigning a firmware to a device: the assigned firmware gets delete after a while - I think only in conjunction when the frontend is active.

Maybe another sign that we need to be careful with keeping cached data.

Reply:
> Yes - I am also not too happy with the global state but did not yet look deeper into it.
> 
> `force_update` likely could be replaced by setting the `update_state` of a device to `registered' or 'finished' and resetting device.fw_version. Maybe it then would be called "reset state".

This is not a bad idea, the idea of force update is that it ignores existing state of the device.  My only concern with this is that if the device sets that version afterwards it will remove the "force", which would happen on either update or reboot of a device.  Update is probably fine, but reboot isn't great.  

Maybe another idea is to extend `UpdateModeEnum` with a `FORCE` mode that always attempts to re-install the currently running firmware (both upgrades and downgrades should be able to be handled with existing flags), and it just sets it to pinned right after?